extensions/classifiers/LDAStringClassifier/LDAStringClassifier.py

```diff
@@ -597,3 +598,91 @@ def predict(self, predicting_data):
         """
         doc_ids = self._add_docs_for_predicting(predicting_data)
         return [self._predict_label_for_doc(doc_id) for doc_id in doc_ids]
+
+    def validate(self, classifier_data):
+        """Validates cached classifier data.
+
+        Args:
+            classifier_data: dict. Consists of various parameter of the
+                classifier.
+        """
+
```

Comment:
Check that keys match, first, before checking each key. Raise a useful error message about missing keys.